coap-gateway/service/client.go

```diff
@@ -79,10 +78,11 @@ type Client struct {
 	observedResourcesLock sync.Mutex
 
 	resourceSubscriptions *kitSync.Map // [token]
-	deviceSubscriptions   *kitSync.Map // [token]
+	//deviceSubscriptions   *kitSync.Map // [token]
 
-	mutex   sync.Mutex
-	authCtx *authCtx
+	mutex                        sync.Mutex
+	authCtx                      *authCtx
+	deviceSubscriptionCancelFunc func(ctx context.Context) error
```

Comment:
Put cancel at the beginning. cancelDeviceSubscription. And is suffix func needed?